Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -123,47 +107,38 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         self.inputStream = inputStream
         self.outputStream = outputStream
         self.logger = logger
+
+        // The stream is configured to deliver its callbacks on the dispatch queue.
+        // This precludes the need for a Thread with RunLoop.
+        CFWriteStreamSetDispatchQueue(outputStream, queue)
     }
 
     // MARK: - Opening & closing
 
-    /// Schedule the output stream on the special thread reserved for stream callbacks.
+    /// Schedule the output stream on the queue for stream callbacks.
     /// Do not wait to complete opening before returning.
     func open() async {
         await withCheckedContinuation { continuation in
-            Self.queue.async {
-                self.perform(#selector(self.openOnThread), on: Self.thread, with: nil, waitUntilDone: false)
+            queue.async {
+                self.outputStream.delegate = self
+                self.outputStream.open()
+                continuation.resume()
             }
-            continuation.resume()
         }
     }
 
-    /// Configure the output stream to make StreamDelegate callback to this bridge using the special thread / run loop, and open the output stream.
-    /// The input stream is not included here.  It will be configured by `URLSession` when the HTTP request is initiated.
-    @objc private func openOnThread() {
-        outputStream.delegate = self
-        outputStream.schedule(in: RunLoop.current, forMode: .default)
-        outputStream.open()
-    }
```

Comment:
The `openOnThread()` method was collapsed into the `open()` method above, since I no longer need to separate the specific steps that are performed on the thread.